### PR TITLE
1 fix issue of  https://github.com/kubeovn/kube-ovn/issues/1151

### PR DESCRIPTION
**Pull request by sxilinux**
What type of this PR

    Bug fixes
  
   I tried to reimplement the func of ovn-leader checking. But I do not know if it is as your expectation.
   would you give me  some comments?

Which issue(s) this PR fixes:
https://github.com/kubeovn/kube-ovn/issues/1151
Fixes #(1151)

### CODE REVIEW

**Comment by oilbeater** [comment]
@sxilinux the reason that we want to move it to go binary is that the `ovn-is-leader.sh` now is doing lots more than just readinessProbe. The leader detection and label pods work should be run in the main cmd rather than a readinessProbe.

What we need is to move this part of work to a new cmd and simplify the readinessProbe.

**Comment by sxilinux** [comment]
@oilbeater , thanks for your reply.
 accodring to your description, my understanding is like that:  
1 replace old new ovn_is _checker.sh by using go cmd( for convenience, we call the go binary as ovn_leader, ovn_leader does the same job as  ovn_is _checker.sh ). and at pod readiness process, insteadlly, we use  ovn_leader.    

am i right?

**Comment by oilbeater** [comment]
@sxilinux `ovn-leader` 需要完成 leader 判断和切换以及 compact 相关的周期性任务，readiness 只需要完成检查 nothd，ovnnb 和 ovnsb 状态的任务即可，相当于把真正的健康检查工作和周期性的处理工作分开。周期性的工作放在 go binary 里，让然后容器启动后最后就执行这个二进制文件即可。健康检查可以还用之前的脚本，但只保留最基础的检查操作，或者将这部分工作也转移到 go 代码里，提供 http 接口进行检查

**Comment by sxilinux** [comment]
@oilbeater ，需求我理解了。我重新把代码调整下。

**Comment by sxilinux** [comment]
@oilbeater  i have remodified the code like what you said. would you review the codes?

**Comment by sxilinux** [comment]
1 i constructed kind-env in my local machine , did the test and  fixed code miss which will cause ci failture.

File: cmd/ovn_leader_checker/ovn_leader_checker.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package ovn_leader_checker
+
+import (
+	"github.com/kubeovn/kube-ovn/pkg/ovn_leader_checker"
+	"os"
+)
+
+func CmdMain() {
+	cfg, err := ovn_leader_checker.ParseFlags()
+	if err != nil {
```
Comment (oilbeater) [inline cmd/ovn_leader_checker/ovn_leader_checker.go:10]: It should run periodically no, not once.

We don't need to exit on error, log the error message is enough.

File: dist/images/start-db.sh [hunk 2]
```diff
@@ -74,6 +74,12 @@ function is_clustered {
   return 1
 }
 
+#### first  try to boot ovn-leader-checker
+#### we put kube-ovn-leader-checker at the same place of ovn-cental
+#### seperate pod of is also ok
+
+/kube-ovn/kube-ovn-leader-checker &
```
Comment (oilbeater) [inline dist/images/start-db.sh:81]: it should replace the last tail command and always run

**Review by oilbeater — CHANGES_REQUESTED** [review]

File: cmd/ovn_leader_checker/ovn_leader_checker.go (hunk 1 above)
Comment (sxilinux) [reply cmd/ovn_leader_checker/ovn_leader_checker.go:10]: i modified codes just like you said.

File: dist/images/start-db.sh (hunk 2 above)
Comment (sxilinux) [reply dist/images/start-db.sh]: i modified codes just like you said.

**Comment by oilbeater** [comment]
Please follow https://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git and squash the 8 commits into one

**Comment by sxilinux** [comment]
ok. i will squash these commits into one

**Comment by sxilinux** [comment]
Hi, [16477a8](https://github.com/kubeovn/kube-ovn/pull/1243/commits/16477a804aeb4351f30d946e27b65ecd8531cb96) has squashed commits into one.

**Comment by oilbeater** [comment]
Thanks! @sxilinux

**Pull request merged**